app/src/main/java/de/westnordost/streetcomplete/overlays/surface/SurfaceColorMapping.kt

```diff
@@ -0,0 +1,82 @@
+ *  - it is OK to use some extra colours if absolutely needed
+ *
+ * - red is reserved for missing data
+ * - black is reserved for generic surface with note tag
+ *
+ * - similar surface should have similar colours
+ *  - all paved ones are one such group
+ *  - all unpaved ones are another
+ *  - grass paver and compacted being high quality unpaved, unhewn cobblestone being low quality unpaved
+ *    should be on border between these two groups
+ *
+ * - asphalt, concrete, paving stones are associated with grey
+ *   but colouring these surfaces as grey resulted in really depressing, unfunny
+ *   and soul-crushing display in cities where everything is well paved.
+ *   Blue is more fun and less sad (while it will not convince anyone
+ *   that concrete desert is a good thing).
+ *
+ * - highly unusual (for roads and paths) surfaces also got black colour
+ * - due to running out of colors all well paved surfaces get the same colour
+ * - due to running out of colours fine gravel, gravel, pebbles, rock got gray surface
+ * - dashes were tested but due to limitation of Tangram were not working well and were
+ *   incapable of replacing colour coding
+ *
+ * - ideally, sand would have colour close to yellow and grass colour close to green caused
+ *   by extremely strong association between surface and colour
+ */
+val Surface.color get() = when (this) {
+    ASPHALT, CHIPSEAL, CONCRETE, PAVING_STONES, PAVING_STONES_WITH_WEIRD_SUFFIX, BRICK, BRICKS
+                       -> Color.BLUE
+    WOOD, METAL, METAL_GRID
+                       -> Color.SKY
+    CONCRETE_PLATES, CONCRETE_LANES, SETT, COBBLESTONE_FLATTENED
+                       -> Color.CYAN
+    UNHEWN_COBBLESTONE, GRASS_PAVER -> Color.AQUAMARINE
+    COMPACTED, FINE_GRAVEL
+                       -> Color.TEAL
+    SAND               -> Color.ORANGE
+    GRASS              -> Color.LIME
+    DIRT, SOIL, EARTH, MUD, GROUND, WOODCHIPS
+                       -> Color.GOLD
+    GRAVEL, PEBBLES, ROCK
+                       -> Color.GRAY
+    CLAY, ARTIFICIAL_TURF, TARTAN
+                       -> Color.BLACK // not encountered in normal situations, get the same as surface with surface:note
+    UNKNOWN, PAVED, UNPAVED
+                       -> Color.BLACK
+}
+
+fun SurfaceAndNote?.getColor(element: Element): String =
+    if (this?.isComplete != true) {
+        // not set but indoor, private or just a "virtual" link -> do not highlight as missing
+        if (isIndoor(element.tags) || isPrivateOnFoot(element) || isLink(element.tags)) {
+            Color.INVISIBLE
+        } else {
+            Color.DATA_REQUESTED
+        }
+    } else {
+        surface!!.color
+    }
+
+private fun isIndoor(tags: Map<String, String>): Boolean = tags["indoor"] == "yes"
+
+private fun isLink(tags: Map<String, String>): Boolean =
+    listOfNotNull(
+        tags["path"],
+        tags["footway"],
+        tags["cycleway"],
+        tags["bridleway"]
+    ).any { it == "link" }
```

Comment:
How often will this be called? If it's performance-critical, should we rewrite it to the (equally well readable) explicit condition?

```suggestion
private fun isLink(tags: Map<String, String>): Boolean =
    tags["path"] == "link"
    || tags["footway"] == "link"
    || tags["cycleway"] == "link"
    || tags["bridleway"] == "link"
```

If it's not performance-critical, should we rewrite it like this to make it more concise (and save one iteration over the whole array)?

```suggestion
private fun isLink(tags: Map<String, String>): Boolean =
    listOf("path", "footway", "cycleway", "bridleway").any { tags[it] == "link" }
```